tools/disasm/disasm.py

```diff
@@ -2,126 +2,260 @@
+    if (
+        symbol == 0x809C46F0
+    ):  # ovl_En_Encount4 fake symbol at the very end of the data section
         return variables_ast[0x809C46DC][0] + " + 0x14"
-    elif symbol == 0x801EF66D: # z_message_nes constant-folding stray fairy array
+    elif symbol == 0x801EF66D:  # z_message_nes constant-folding stray fairy array
         return variables_ast[0x801EF670][0] + f" - 0x{0x801EF670 - 0x801EF66D:X}"
-    elif symbol == 0x80A09740: # boss_07 symbol with large addend folded into %lo
+    elif symbol == 0x80A09740:  # boss_07 symbol with large addend folded into %lo
         return variables_ast[0x80A09A60][0] + f" - 0x{0x80A09A60 - 0x80A09740:X}"
-    elif symbol == 0x80B80248: # bg_ikana_mirror symbol with large addend folded into %lo
+    elif (
+        symbol == 0x80B80248
+    ):  # bg_ikana_mirror symbol with large addend folded into %lo
         return variables_ast[0x80B801A8][0] + f" + 0x{0x80B80248 - 0x80B801A8:X}"
-    elif symbol == 0x8084D2FC: # player symbol with very large addend folded into %lo, since we don't know the real symbol just use the first data symbol for now
+    elif (
+        symbol == 0x8084D2FC
+    ):  # player symbol with very large addend folded into %lo, since we don't know the real symbol just use the first data symbol for now
         return variables_ast[0x8085B9F0][0] + f" - 0x{0x8085B9F0 - 0x8084D2FC:X}"
-    elif symbol == 0x001ABAB0 or symbol == 0x001E3BB0: # OS_K0_TO_PHYSICAL on rspS2DEX text and data symbols
+    elif (
+        symbol == 0x001ABAB0 or symbol == 0x001E3BB0
+    ):  # OS_K0_TO_PHYSICAL on rspS2DEX text and data symbols
         return variables_ast[symbol + 0x80000000][0] + " - 0x80000000"
-    elif symbol == 0x00AC0480: # do_action_static + 0x480, this is the only rom segment that has a constant offset folded into it so just hack it
+    elif (
+        symbol == 0x00AC0480
+    ):  # do_action_static + 0x480, this is the only rom segment that has a constant offset folded into it so just hack it
         return "_do_action_staticSegmentRomStart + 0x480"
 
     # real names
-    if symbol in functions and symbol in functions_ast.keys():
+    if symbol in functions_ast.keys():
         return functions_ast[symbol][0]
     elif symbol in variables_ast.keys():
         return variables_ast[symbol][0]
-    elif symbol in [addr for _,addr in vrom_variables]:
+    elif symbol in [addr for _, addr in vrom_variables]:
         # prefer "start" vrom symbols
-        if symbol in [addr for name,addr in vrom_variables if "SegmentRomStart" in name]:
-            return [name for name,addr in vrom_variables if "SegmentRomStart" in name and addr == symbol][0]
+        if symbol in [
+            addr for name, addr in vrom_variables if "SegmentRomStart" in name
+        ]:
+            return [
+                name
+                for name, addr in vrom_variables
+                if "SegmentRomStart" in name and addr == symbol
+            ][0]
```

Comment:
This bit looks a bit wasteful, since it is doing the same operation twice
I think using a temp would benefit performance a bit
Something like:
```suggestion
        filteredVromSymbols = { addr: name for name, addr in vrom_variables if "SegmentRomStart" in name}
        if symbol in filteredVromSymbols:
            return filteredVromSymbols[symbol]
```

Reply:
It's not quite the same thing twice. First it checks if if the name is in the list, then it returns if name is in and the addr == symbol. I dunno what this code does really so I don't know if that will break things.